src/show_message.rs

```diff
@@ -0,0 +1,137 @@
+use std::borrow::Cow;
+
+use itertools::Itertools;
+use jsonrpc_core::{Id, MethodCall};
+use lsp_types::{MessageActionItem, MessageType, ShowMessageParams, ShowMessageRequestParams};
+use serde::Deserialize;
+
+use crate::{context::Context, types::EditorMeta, util::editor_quote};
+
+/// Queues the message request from the LSP server.
+pub fn show_message_request(meta: EditorMeta, request: MethodCall, ctx: &mut Context) {
+    let params: ShowMessageRequestParams = request
+        .params
+        .clone()
+        .parse()
+        .expect("Failed to parse ShowMessageRequest params");
+    ctx.pending_message_requests.push_back((request.id, params));
+    update_modeline(meta, ctx)
+}
+
+#[derive(Deserialize)]
+struct MessageRequestResponse {
+    pub message_request_id: Option<Id>,
+    pub item: Option<toml::Value>,
+}
+
+/// Handles an user's response to a message request (or the user's request to display the next message request).
+pub fn show_message_request_respond(meta: EditorMeta, params: toml::Value, ctx: &mut Context) {
+    let resp =
+        MessageRequestResponse::deserialize(params).expect("Cannot parse message request response");
+    let req_id = match resp.message_request_id {
+        None => return show_message_request_next(meta, ctx),
+        Some(id) => id,
```

Comment:
can write this as

	let Some(req_id) = resp.message_request_id else { return show_message_request_next(meta, ctx); }

but actually I don't think it's great to overload `window/showMessageRequest` with two disjoint meanings

Make it two separate methods instead?

- window/showMessageRequest/showNext
- window/showMessageRequest/respond